### DISCUSSION TITLE
is it RabbitMQ is good tool to use it in a chat app for queue the message data?

### DISCUSSION THREAD

**Discussion by AbobakarSadeeq** [rabbitmq/rabbitmq-server#11227] [Questions] [answered]
is it RabbitMQ is right tool for me to use it in a chat app for to queue the actual message data of sended user to offline user so, user dont miss any kind of sended message from other user while user is offline?

is it RabbitMQ can handle the multiple user i mean thousands of user for to store their queue message?

**Comment by michaelklishin** [comment, marked as answer]
Thousands of users are not going to be a problem, and neither will [thousands of concurrent connections](https://www.rabbitmq.com/blog/2023/03/21/native-mqtt) (even if beyond a certain number of them you'd have to [start tweaking certain defaults](https://www.rabbitmq.com/docs/networking#tuning-for-large-number-of-connections-tcp-buffer-size)).

As explained on Discord, using a combination of non-replicated classic queues (lower footprint per queue), quorum queues (data safety and replication) and streams (a different data structure with its own [pros and cons](https://www.rabbitmq.com/docs/streams)) as the messaging layer for a chat app sounds very reasonable. You'd need to use a suitable data store for queries and long term storage, though.
